### PR TITLE
Patch for Go mobile on recent docker versions

### PR DESCRIPTION
**Pull request by xiam**
See: https://github.com/getlantern/lantern/issues/2552

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,7 +34,11 @@ RUN (curl -sSL $GO_PACKAGE_URL | tar -xvz -C /tmp) && \
 RUN mkdir -p $GOROOT && \
   git clone https://go.googlesource.com/go $GOROOT && \
   cd $GOROOT && \
-  git checkout -b go1.4 origin/release-branch.go1.4
+  git checkout release-branch.go1.4
+
+# Patch for skipping a failing test in new docker versions. See https://github.com/getlantern/lantern/issues/2578
+RUN yum install -y patch && yum clean all
+RUN curl https://gist.githubusercontent.com/xiam/f50f6dd6085f9a07ccfd/raw/5e0f472221f9c1556fe34788ff01724b63980337/docker_golang | patch -p0
```
Comment (xiam) [inline Dockerfile]: The same patch I used to fix this one: https://github.com/getlantern/lantern/issues/2578

File: Makefile [hunk 2]
```diff
@@ -28,7 +28,7 @@ PACKAGE_URL := https://www.getlantern.org
 
 LANTERN_BINARIES_PATH ?= ../lantern-binaries
 
-GO_MOBILE_REVISION=754ad3cfda342ad8c5d07a5ef9334f8a62b6aefa
+GO_MOBILE_REVISION=47553f4d4275316f2610b3204f8f9ecaf3ce63a7
```
Comment (xiam) [inline Makefile]: A revision from April, around the time I first commited the change that introduced the `libgojni.so` compilation from the `Makefile`.

File: Makefile [hunk 3]
```diff
@@ -197,7 +199,9 @@ docker-golang-android: require-mercurial
 		$(GO) get -d golang.org/x/mobile/example/... && \
 		$(GO) get golang.org/x/mobile/cmd/gobind && \
 		cd src/golang.org/x/mobile && \
+		git reset --hard && \
 		git checkout $(GO_MOBILE_REVISION) && \
+		curl https://gist.githubusercontent.com/xiam/56d8fe575176f459613d/raw/8873ed3af55a7d449a1c9c2112de118da968dbc3/gistfile1.txt | git apply - && \
```
Comment (xiam) [inline Makefile]: The same patch I used to fix this one: #2578

File: Makefile [hunk 4]
```diff
@@ -425,7 +429,12 @@ android-lib: docker-golang-android
 		gobind -lang=go github.com/getlantern/lantern-android/libflashlight/bindings > bindings/go_bindings/go_bindings.go && \
 		gobind -lang=java github.com/getlantern/lantern-android/libflashlight/bindings > bindings/Flashlight.java || exit 1;
 	@$(DOCKER) run -v $$PWD/src:/src golang/mobile /bin/bash -c \ "cd /src/github.com/getlantern/lantern-android/libflashlight && ./make.bash" && \
-	ls -l src/github.com/getlantern/lantern-android/app/libs/armeabi-v7a/libgojni.so
+	ls -l src/github.com/getlantern/lantern-android/app/libs/armeabi-v7a/libgojni.so && \
+	if [ -d "$(FIRETWEET_DIR)" ]; then \
```
Comment (xiam) [inline Makefile]: Copying to `FIRETWEET_DIR`. If such env value was given:

```
FIRETWEET_DIR=/path/to/firetweet make android-lib
```

**Comment by xiam** [comment]
This should be reviewed after merging https://github.com/getlantern/lantern/pull/2583

File: Makefile (hunk 2 above)
Comment (atavism) [inline Makefile]: Great job hunting that down!

**Comment by atavism** [comment]
This works great for me. Merging.

**Pull request merged**